go/core.rst

```diff
@@ -0,0 +1,425 @@
+Core go rules
+=============
+
+.. _test_filter: https://bazel.build/versions/master/docs/bazel-user-manual.html#flag--test_filter
+.. _test_arg: https://bazel.build/versions/master/docs/bazel-user-manual.html#flag--test_arg
+.. _gazelle: tools/gazelle/README.md
+.. _build constraints: http://golang.org/pkg/go/build/
+.. _GoLibrary: providers.rst#GoLibrary
+.. _GoEmbed: providers.rst#GoEmbed
+.. _GoBinary: providers.rst#GoBinary
+.. _cgo: http://golang.org/cmd/cgo/
+.. _"Make variable": https://docs.bazel.build/versions/master/be/make-variables.html
+.. _Bourne shell tokenization: https://docs.bazel.build/versions/master/be/common-definitions.html#sh-tokenization
+.. _data dependencies: https://docs.bazel.build/versions/master/build-ref.html#data
+.. _static: modes.rst#using-the-race-detector
+.. _race: modes.rst#building-static-binaries
+
+.. role:: param(kbd)
+.. role:: type(emphasis)
+.. role:: value(code)
+.. |mandatory| replace:: **mandatory value**
+
+These are the core go rules, required for basic operation.
+The intent is that theses rules are sufficient to match the capabilities of the normal go tools.
+
+.. contents::
+
+-----
+
+Design
+------
+
+Defines and stamping
+~~~~~~~~~~~~~~~~~~~~
+
+**TODO**: More information
+
+Embedding
+~~~~~~~~~
+
+This is used for things like internal tests, where a library is recompiled with additional
+and also code generators where the generated source will be known to have extra dependancies.
+
+**TODO**: More information
+
+API
+---
+
+go_library
+~~~~~~~~~~
+
+This builds a Go library from a set of source files that are all part of
+the same package.
+
+Providers
+^^^^^^^^^
+
+* GoLibrary_
+* GoEmbed_
+
+Output groups
+^^^^^^^^^^^^^
+
+* default *: A library with the default build options.*
```

Comment:
Would be better to have the group names emphasized and the descriptions in normal face.

Reply:
Done